.github/workflows/ci.yml

```diff
@@ -9,36 +9,32 @@ on:
 
 jobs:
   check:
-    name: check
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        build: [msrv, stable]
         features:
           [
             "",
             "--no-default-features",
             "--no-default-features --features use_alloc",
             "--all-targets --all-features",
           ]
-        include:
-          - build: msrv
-            rust: 1.62.1
-          - build: stable
-            rust: stable
-        exclude:
-          - build: msrv
-            # we only care about the MSRV with respect to the lib target
-            features: "--all-targets --all-features"
     steps:
       - uses: actions/checkout@v4
-      - uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: ${{ matrix.rust }}
+      - uses: dtolnay/rust-toolchain@stable
       - run: cargo check ${{ matrix.features }}
 
+  msrv:
+    runs-on: ubuntu-latest
+    env:
+      CARGO_NET_GIT_FETCH_WITH_CLI: true
```

Comment:
the 1.43.1 version of cargo does not appear to work in CI without this line. This should be reviewed on the next MSRV bump